### PR TITLE
pipeline: init at 1.15.0

### LINKED ISSUE #331023: Package request: pipeline
**Project description**

Pipeline lets you watch and download videos from YouTube and PeerTube, all without needing to navigate through different websites.

**Metadata**

* homepage URL: https://gitlab.com/schmiddi-on-mobile/pipeline
* source URL: https://gitlab.com/schmiddi-on-mobile/pipeline
* license: gpl3+
* platforms: linux

### PR DESCRIPTION
**Pull request by ByteSudoer**
## Description of changes
Pipeline lets you watch and download videos from YouTube and PeerTube, all without needing to navigate through different websites.

Metadata
- homepage URL: https://gitlab.com/schmiddi-on-mobile/pipeline
- source URL: https://gitlab.com/schmiddi-on-mobile/pipeline
- license: gpl3+

Closes https://github.com/NixOS/nixpkgs/issues/331023

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  0- [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [X] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/pi/pipeline/package.nix [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    gdk-pixbuf
+    glib
+    gtk4
+    libadwaita
+    openssl
+    pango
+  ] ++ lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.Security ];
+
+  postInstall = ''
+    mkdir -p $out/bin
```
Comment (AndersonTorres) [inline pkgs/by-name/pi/pipeline/package.nix:72]: Why creating a directory here?

Comment (ByteSudoer) [reply pkgs/by-name/pi/pipeline/package.nix:72]: The output binary is different from the mainProgram so i symlinked to match the pname. But you are right that can be done wihtout creating the directory.

**Review by felbinger — APPROVED** [review]

File: pkgs/by-name/pi/pipeline/package.nix [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    meson
+    gettext
+    glib
+    pkg-config
+    desktop-file-utils
+    appstream
+    ninja
+    rustc
+    cargo
+    openssl
```
Comment (SuperSandro2000) [inline pkgs/by-name/pi/pipeline/package.nix:57]: Why do we need openssl in nativeBuildInputs?

File: pkgs/by-name/pi/pipeline/package.nix [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+  nativeBuildInputs = [
+    wrapGAppsHook4
+    rustPlatform.cargoSetupHook
+    meson
+    gettext
+    glib
+    pkg-config
+    desktop-file-utils
+    appstream
```
[the remark this answers is not in the document]
Comment (ByteSudoer) [reply pkgs/by-name/pi/pipeline/package.nix:53]: yes. And it's needed as the build fails without it

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/by-name/pi/pipeline/package.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/by-name/pi/pipeline/package.nix:57]: That doesn't answer my question. Their flake can also be wrong which I think it is. We usually don't want to use openssl-config but pkg-config and besides the openssl cli I don't think there are other executables in the package.

I tried building the package locally and openssl was not required here.

Comment (SuperSandro2000) [reply pkgs/by-name/pi/pipeline/package.nix:57]:
```suggestion
```

File: pkgs/by-name/pi/pipeline/package.nix (hunk 3 above)
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/by-name/pi/pipeline/package.nix:53]: Libraries belong into buildInputs, not nativeBuildInputs.

Please move gettext, glib, appstream and openssl (which is already there9 to buildInputs.

**Comment by drupol** [comment]
@ByteSudoer Do you plan to update your PR at some point? I'm going to move it to draft in the meantime, feel free to mark it as ready when you feel it's ready to be reviewed again.

**Pull request closed without merging**